packages/@aws-cdk/aws-eks/lib/cluster-resource.ts

```diff
@@ -32,51 +30,66 @@ export class ClusterResource extends Construct {
    * that gets administrator privilages on the cluster (`system:masters`), and
    * will be able to issue `kubectl` commands against it.
    */
-  public readonly creationRole: iam.IRole;
+  private readonly creationRole: iam.Role;
+  private readonly trustedPrincipals: string[] = [];
 
   constructor(scope: Construct, id: string, props: CfnClusterProps) {
```

Comment:
This accepts `CfnClusterProps`, makes it a very leaky abstraction, and I'd argue this is not ideal. Especially not since you don't control that API and CloudFormation can shamelessly break it under you.

Reply:
This is intentional. I wanted the custom resource (which is not publically exported) to have the exact same API as the stock CFN resource.

Reply:
Okay - but then this class' stability _must_ be the same as the stock CFN resource's (aka `@stability external`).

Reply:
This is an internal unexported class. There is no meaning to it’s @stability annotation.